### DISCUSSION TITLE
Syntax highlight doesn't work properly.

### DISCUSSION THREAD

**Discussion by pannchat** [mdx-js/mdx#1874] [Q&A] [answered]
```js
import Content from '!babel-loader!@mdx-js/loader!./content.mdx'

export default function App() {
  return <Content components={{code}}/>
}

function code({className, ...props}) {
  alert("s");
  const match = /language-(\w+)/.exec(className || '')
  return match
    ? <SyntaxHighlighter language={match[1]} PreTag="div" {...props} />
    : <code className={className} {...props} />
}
```

I made this code by referring to the official document.
But `components={{}}` It doesn't seem to work.
@mdx-js/loader": "^1.6.22
 "react": "^17.0.2"

**Comment by wooorm** [comment, marked as answer]
See the note at the top of the website: it’s for MDX 2, which you can install with `@next`

**Reply by pannchat** [reply]
Thank you so much. But the next version is...
```
TypeError: this.getOptions is not a function
At Object.loader (node_modules/@mdx-js/loader/lib/index.js:23:55) is caused by a user.
```
I used the lower version. (i used CRA)

**Reply by wooorm** [reply]
Old docs: https://github.com/mdx-js/mdx/blob/master/docs/guides/syntax-highlighting.mdx.

**Reply by pannchat** [reply]
ah..! omg You've saved my life!!  Have a good day.
